src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -184,6 +228,34 @@ func (l *commitLog) Open() error {
 	return nil
 }
 
+func (l *commitLog) ActiveLogs() ([]File, error) {
+	l.closedState.Lock()
+	defer l.closedState.Unlock()
+
+	if l.closedState.closed {
+		return nil, errCommitLogClosed
+	}
+
+	var (
+		err  error
+		file File
+		wg   = sync.WaitGroup{}
+	)
+	wg.Add(1)
+
+	l.writes <- commitLogWrite{
+		valueType: activeLogsValueType,
+		valueTypeFn: func(f File, e error) {
+			err = e
+			file = f
+			wg.Done()
+		},
+	}
+
+	wg.Wait()
+	return []File{file}, err
```

Comment:
Do you want to bother doing the `if err != nil { return nil, err}; return []File{file}, nil` so you only return something if and only if `err == nil`?  Just slightly more idiomatic Go to do so.

Reply:
yeah thats fair, Its kind of messed up to return a non-empty slice in the error case